card/lib/card/act_manager.rb

```diff
@@ -154,17 +157,43 @@ def deep_delete director
       def running_act?
         (dir = act_director) && dir.running?
       end
+      
+      # If active jobs (and hence the integrate_with_delay events) don't run
+      # in a background process then Card::Env.deserialize! decouples the
+      # controller's params hash and the Card::Env's params hash with the
+      # effect that params changes in the CardController get lost
+      # (a crucial example are success params that are processed in
+      # CardController#update_params_for_success)
+      def contextualize act_id, card, env, auth
+        if delaying?
+          contextualize_for_delay(act_id, card, env, auth) { yield }
```

Comment:
rationale is that there are LOTS of things (env, auth, act, directors, clearing...) that need to happen if delayed but not if happening inline.  This harder separation cleans that up.